tests/invest/flaskapp.test.js

```diff
@@ -39,140 +43,153 @@ afterAll(async () => {
   await shutdownPythonProcess();
 });
 
-test('invest list items have expected properties', async () => {
-  const investList = await server_requests.getInvestModelNames();
-  Object.values(investList).forEach((item) => {
-    expect(item.model_name).not.toBeUndefined();
+describe('requests to flask endpoints', () => {
+  let WORKSPACE;
+  beforeAll(() => {
+    WORKSPACE = fs.mkdtempSync(path.join(os.tmpdir(), 'data-'));
   });
-});
 
-test('fetch invest model args spec', async () => {
-  const spec = await server_requests.getSpec('carbon');
-  const expectedKeys = ['model_name', 'pyname', 'userguide_html', 'args'];
-  expectedKeys.forEach((key) => {
-    expect(spec[key]).not.toBeUndefined();
+  afterAll(() => {
+    rimraf.sync(WORKSPACE);
   });
-});
 
-test('fetch invest validation', async () => {
-  const spec = await server_requests.getSpec('carbon');
-  // it's okay to validate even if none of the args have values yet
-  const argsDict = argsDictFromObject(spec.args);
-  const payload = {
-    model_module: spec.pyname,
-    args: JSON.stringify(argsDict),
-  };
-
-  const results = await server_requests.fetchValidation(payload);
-  // There's always an array of arrays, where each child array has
-  // two elements: 1) an array of invest arg keys, 2) string message
-  expect(results[0]).toHaveLength(2);
-});
+  test('invest list items have expected properties', async () => {

```

Comment:
Hmm the line-by-line diff of this file is not so useful because most of the changes are just moving blocks of code around, moving tests into a `describe` block.